### ISSUE TITLE
virtualbox-ovf provisioner lowercases checksum file paths

### ISSUE THREAD

**Issue by tardfree** [hashicorp/packer#9049]
#### Overview of the Issue
I recently started switching from checksums in packer json files to using checksum files and noticed this problem. The checksum parameter seems to be lowercased in the virtualbox-ovf builder, leading to file not found issues.
I've created a small example using a publically available ova file (though any ova should be fine as it doesn't need to boot, only import to vbox).

#### Reproduction Steps
Test setup: download https://downloads.vyos.io/release/legacy/1.1.8/vyos-1.1.8-amd64.ova and save it with a captial V to /tmp/Vyos-1.1.8-amd64.ova 

Create a checksum file: sha256sum Vyos-1.1.8-amd64.ova > Vyos-1.1.8-amd64.ova.sha256

Checksum file contents: 9e8ab744328011e254fbbf12fae7ce9806a8e80c8b295329fdb77101ff0dfc84  Vyos-1.1.8-amd64.ova

Test file:
```
{
  "builders": [
    {
      "source_path": "/tmp/Vyos-1.1.8-amd64.ova",
      "checksum_type": "file",
      "checksum": "/tmp/Vyos-1.1.8-amd64.ova.sha256",

      "type": "virtualbox-ovf",
      "guest_additions_mode": "disable",
      "ssh_username" : "root",
      "headless": false,
      "keep_registered": false
    }
  ]
}
```
Produces (Vyos-test1 in gist below)
```
==> virtualbox-ovf: Retrieving OVF/OVA
==> virtualbox-ovf: Trying /tmp/Vyos-1.1.8-amd64.ova
==> virtualbox-ovf: Trying /tmp/Vyos-1.1.8-amd64.ova?checksum=file%3A%2Ftmp%2Fvyos-1.1.8-amd64.ova.sha256
==> virtualbox-ovf: Download failed invalid checksum: Error downloading checksum file: source path error: stat /tmp/vyos-1.1.8-amd64.ova.sha256: no such file or directory
==> virtualbox-ovf: error downloading OVF/OVA: [invalid checksum: Error downloading checksum file: source path error: stat /tmp/vyos-1.1.8-amd64.ova.sha256: no such file or directory]
==> virtualbox-ovf: Deleting output directory...
Build 'virtualbox-ovf' errored: error downloading OVF/OVA: [invalid checksum: Error downloading checksum file: source path error: stat /tmp/vyos-1.1.8-amd64.ova.sha256: no such file or directory]
```
If you rename the /tmp/Vyos-1.1.8-amd64.ova.sha256 to /tmp/vyos-1.1.8-amd64.ova.sha256  (lower case first letter) and build the same json file, it now succeeds.

Output when checksum is renamed to lowercase (same json file which references the upper cased name)  (Vyos-test2 in gist below)
```
==> virtualbox-ovf: Retrieving OVF/OVA
==> virtualbox-ovf: Trying /tmp/Vyos-1.1.8-amd64.ova
==> virtualbox-ovf: Trying /tmp/Vyos-1.1.8-amd64.ova?checksum=file%3A%2Ftmp%2Fvyos-1.1.8-amd64.ova.sha256
==> virtualbox-ovf: /tmp/Vyos-1.1.8-amd64.ova?checksum=file%3A%2Ftmp%2Fvyos-1.1.8-amd64.ova.sha256 => /tmp/Vyos-1.1.8-amd64.ova
==> virtualbox-ovf: Importing VM: /tmp/Vyos-1.1.8-amd64.ova
==> virtualbox-ovf: Creating forwarded port mapping for communicator (SSH, WinRM, etc) (host port 2706)
==> virtualbox-ovf: Starting the virtual machine...
==> virtualbox-ovf: Waiting 10s for boot...
```
note the 3rd and 4th line there shows the lower case v in the checksum filename, even though the json file has upper case. The build succeeds becasue we renamed the file.

I've also tried checksum URI's like `file:/tmp/Vyos-1.1.8-amd64.ova.sha256` and `file:///tmp/Vyos-1.1.8-amd64.ova.sha256` which made no difference.

#### Workaround
I found a workaround. If I specify the source_path with the checksum on the end (go-getter format), it respects the strings case.

Test file (rename the checksum file back to uppercase first letter before running)
```
{
  "builders": [
    {
      "source_path": "/tmp/Vyos-1.1.8-amd64.ova?checksum=/tmp/Vyos-1.1.8-amd64.ova.sha256",
      "checksum_type": "file",

      "type": "virtualbox-ovf",
      "guest_additions_mode": "disable",
      "ssh_username" : "root",
      "headless": false,
      "keep_registered": false
    }
  ]
}
```
does work (Vyos-test3 in gist below)
```
==> virtualbox-ovf: Retrieving OVF/OVA
==> virtualbox-ovf: Trying /tmp/Vyos-1.1.8-amd64.ova?checksum=/tmp/Vyos-1.1.8-amd64.ova.sha256
==> virtualbox-ovf: Trying /tmp/Vyos-1.1.8-amd64.ova?checksum=file%3A%2Ftmp%2FVyos-1.1.8-amd64.ova.sha256
==> virtualbox-ovf: /tmp/Vyos-1.1.8-amd64.ova?checksum=file%3A%2Ftmp%2FVyos-1.1.8-amd64.ova.sha256 => /tmp/Vyos-1.1.8-amd64.ova
==> virtualbox-ovf: Importing VM: /tmp/Vyos-1.1.8-amd64.ova
==> virtualbox-ovf: Creating forwarded port mapping for communicator (SSH, WinRM, etc) (host port 2566)
==> virtualbox-ovf: Starting the virtual machine...
==> virtualbox-ovf: Waiting 10s for boot...
```
note the 3rd and 4th line there shows the upper case V in the checksum filename.

### Packer version
* Packer v1.5.5
### Operating system and Environment details
* Fedora 31
* Virtualbox 6.1.4

### Log Fragments and crash.log files
PACKER_LOG=1 runs of the 3 tests above are in this [gist](https://gist.github.com/tardfree/f7d41f1d5c1b36e084a990b59cb58e40)

**Comment by nywilken** [comment]
Hi @tardfree thanks for reaching out. The issue lies in the Prepare method of the builder https://github.com/hashicorp/packer/blob/31946d551e9aefb13cfca5c044d4e389f5e79808/builder/virtualbox/ovf/config.go#L158

It's possible that this was being checked against as some previous point. The workaround is a good way forward for the moment as specifying it as part of the `source_url` ensures it gets passed directly to go-getter unmodified https://github.com/hashicorp/packer/blob/31946d551e9aefb13cfca5c044d4e389f5e79808/common/step_download.go#L114

**Comment by ryanhartje** [comment]
Hi @nywilken , I see your point but I'm having a hard time understanding how to setup a test such as this. Do you perhaps know of an example I could reference that models how I should test this?

**Comment by nywilken** [comment]
Hi @ryanhartje apologies I missed your comment. I’ll take a look at your PR, most likely tomorrow, and the underlying code to see how to best test. Thanks for taking on this issue.

**Comment by nywilken** [comment]
Closed by #9101
